hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/s3a/ITestS3GuardOutOfBandOperations.java

```diff
@@ -969,16 +970,42 @@ private void deleteFileInListing()
       deleteFile(rawFS, testFilePath);
 
       // File status will be still readable from s3guard
-      FileStatus status = guardedFs.getFileStatus(testFilePath);
+      S3AFileStatus status = (S3AFileStatus)
+          guardedFs.getFileStatus(testFilePath);
       LOG.info("authoritative: {} status: {}", allowAuthoritative, status);
-      expectExceptionWhenReading(testFilePath, text);
-      expectExceptionWhenReadingOpenFileAPI(testFilePath, text, null);
-      expectExceptionWhenReadingOpenFileAPI(testFilePath, text, status);
+      if (isVersionedChangeDetection() && status.getVersionId() != null) {
+        // when the status entry has a version ID, then that may be used
+        // when opening the file on what is clearly a versioned store.
+        int length = text.length();
+        byte[] bytes = readOpenFileAPI(guardedFs, testFilePath, length, null);
+        Assertions.assertThat(toChar(bytes))
+            .describedAs("openFile(%s)", testFilePath)
+            .isEqualTo(text);
+        // reading the rawFS with status will also work.
+        bytes = readOpenFileAPI(rawFS, testFilePath, length, status);
```

Comment:
`bytes = readOpenFileAPI(rawFS, testFilePath, length, null);` should still fail right? Do you think we can also that in this `if` clause?

Reply:
`bytes = readOpenFileAPI(rawFS, testFilePath, length, null);` won't fail (tested). Why should it fail? The`FileStatus` is just an additional parameter for the`FutureDataInputStreamBuilder`.

Reply:
> bytes = readOpenFileAPI(rawFS, testFilePath, length, null);

The file was deleted and I was expecting FNF exception here. I have not tested here with versioned  bucket.

Reply:
`bytes = readOpenFileAPI(rawFS, testFilePath, length, null)` should fail; no status coming in, will add the check